python/ray/tune/tests/test_api.py

```diff
@@ -364,69 +363,59 @@ def train(config, reporter):
         )
 
     def testLogdir(self):
-        def train(config, reporter):
-            assert (
-                os.path.join(ray._private.utils.get_user_temp_dir(), "logdir", "foo")
-                in os.getcwd()
-            ), os.getcwd()
-            reporter(timesteps_total=1)
-
-        register_trainable("f1", train)
-        run_experiments(
-            {
-                "foo": {
-                    "run": "f1",
-                    "local_dir": os.path.join(
-                        ray._private.utils.get_user_temp_dir(), "logdir"
-                    ),
-                    "config": {"a": "b"},
-                }
-            }
-        )
+        logdir = os.path.join(ray._private.utils.get_user_temp_dir(), "logdir")
+
+        def train_fn(config):
+            assert logdir in os.getcwd(), os.getcwd()
+            train.report(dict(timesteps_total=1))
+
+        register_trainable("f1", train_fn)
+        with unittest.mock.patch.dict(os.environ, {"TEST_TMPDIR": logdir}):
+            tune.run("f1")
 
     def testLogdirStartingWithTilde(self):
         local_dir = "~/ray_results/local_dir"
 
-        def train(config, reporter):
+        def train_fn(config):
             cwd = os.getcwd()
             assert cwd.startswith(os.path.expanduser(local_dir)), cwd
             assert not cwd.startswith("~"), cwd
-            reporter(timesteps_total=1)
+            train.report(dict(timesteps_total=1))
 
-        register_trainable("f1", train)
-        run_experiments(
-            {
-                "foo": {
-                    "run": "f1",
-                    "local_dir": local_dir,
-                    "config": {"a": "b"},
+        register_trainable("f1", train_fn)
+        with unittest.mock.patch.dict(
+            os.environ, {"TEST_TMPDIR": os.path.expanduser(local_dir)}
```

Comment:
Does expanding the user here make this test case a duplicate of the above?

Reply:
Hm, probably. It actually requires the expansion. Maybe we can just remove it.

Reply:
Removed